salt/modules/powerdns.py

```diff
@@ -0,0 +1,567 @@
+"""
+Create, modify and delete PowerDNS zone
+"""
+
+import logging
+import socket
+
+import requests
+from salt.exceptions import CommandExecutionError
+
+# Constant section
+
+DEFAULT_TTL = 300  # In Seconds
+DEFAULT_NS_TTL = 3600  # In Seconds
+
+REPLACE_CHANGETYPE = "REPLACE"
+DELETE_CHANGETYPE = "DELETE"
+
+DNS_RECORD_NS = "NS"
+DNS_RECORD_SOA = "SOA"
+DNS_RECORD_TXT = "TXT"
+DNS_RECORD_A = "A"
+DNS_RECORD_CNAME = "CNAME"
+
+HTTP_RESPONSE_CODE_UNPROCESSABLE_ENTITY = 422
+HTTP_RESPONSE_CODE_CREATED = 201
+HTTP_RESPONSE_CODE_NO_CONTENT = 204
+
+POWERDNS_URL_ZONES_TEMPLATE = "/api/v1/servers/localhost/zones"
+POWERDNS_AUTH_HEADER = "X-API-Key"
+
+logger = logging.getLogger(__name__)
```

Comment:
```suggestion
log = logging.getLogger(__name__)
```
Please name the logging handler instance `log`, not `logger`.

Reply:
Fixed